app/Models/Multiplayer/ScoreLink.php

```diff
@@ -29,6 +29,10 @@ public static function complete(ScoreToken $token, array $params): static
     {
         return \DB::transaction(function () use ($params, $token) {
             $score = Score::createFromJsonOrExplode($params);
+            
+            // multiplayer scores are always preserved.
+            $score->preserve = true;
+            $score->saveOrExplode();
```

Comment:
This probably looks funny/awkward but there was an intention behind doing things that way. In most cases, scores are created by passing raw json params to `Score::extractParams()` and then passing that to `Score::createFromJsonOrExplode()`. But I didn't want to add a parameter to control `preserve` to `extractParams()` (which I would have had to do to avoid this silliness), because I don't want to allow an arbitrary api consumer to set `preserve` directly in the first place?

If I'm overthinking this do let me know and I'll shuffle across, but yeah that was the intention.

Reply:
seems fine but I think it can also be done with `createFromJsonOrExplode([...$params, 'preserve' => true])`

Reply:
Yeah that works, good call. Applied, thanks :+1: